env0/resource_environment.go

```diff
@@ -296,11 +309,21 @@ func resourceEnvironmentRead(ctx context.Context, d *schema.ResourceData, meta i
 		return diag.Errorf("could not get environment: %v", err)
 	}
 
+	var template client.Template
```

Comment:
if it's not a templateless environment, template will remain an empty struct.
`setEnvironmentSchema` will ignore it in that case...